### ISSUE TITLE
Research the CachePot for the runtime compilation speed up.

### ISSUE THREAD

**Issue by shamil-gadelshin** [Joystream/joystream#3035]
Here is the [video from the Parity Conference](https://www.youtube.com/watch?v=z1-do6e0N90&t=716s&ab_channel=ParityTechnologies).

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201703952879841/1201704774059285) by [Unito](https://www.unito.io)

**Comment by shamil-gadelshin** [comment]
It looks like the developers do not recommend the project yet - they're having issues with the c-bindings and clang compiler (as of the time of the video). [The documentation](https://cachepot.cc/) has a lot of gaps. However, it promises a 40% decrease in the compilation time. 

I would wait for more elaborated versions unless we have urgent performance issues.

**Comment by mnaamani** [comment]
We can look into it further
Another avenue worth re-visiting is some experimental work in: 

https://github.com/Joystream/joystream/blob/development/devops/dockerfiles/node-and-runtime/Dockerfile_experimental

TLDR: mount cargo cache and artifacts from local machine in the docker build to speedup builds builds. Things to watch out for, make sure docker build puts different arch builds in correct 'target' folder so docker builds don't interference with host machine native builds..

cc: @ahhda
